payu/calendar.py

```diff
@@ -90,6 +90,68 @@ def calculate_leapdays(init_date, final_date):
     return datetime.timedelta(days=leap_days)
 
 
+# TODO: The caltype logic could be simplified if we switched
+# to using just a string as the caltype input. Might require reworking of other
+# functions for consistency.
+def seconds_between_dates(start_date, end_date, caltype_int):
+    """
+    Calculate the number of seconds between two datetime objects
+    with a specified calender type by using cftime datetime objects
+    as intermiaries.
+
+    Parameters
+    ----------
+    start_date: datetime.date
+    end_date: datetime.date
+    caltype: Integer, either GREGORIAN or NOLEAP
+
+    Returns
+    -------
+    seconds: Number of seconds between start_date and end_date.
+    """
+    # Get the cftime string corresponding to the caltype integer
+
+    # TODO: Is it confusing that GREGORIAN means proleptic gregorian?
+    if caltype_int == GREGORIAN:
+        calendar_str = "proleptic_gregorian"
+    elif caltype_int == NOLEAP:
+        calendar_str = "noleap"
+    else:
+        raise ValueError(f"Unrecognized caltype integer {caltype_int}")
+
+    delta = (date_to_cftime(end_date, calendar_str)
+             - date_to_cftime(start_date, calendar_str))
+
+    return delta.total_seconds()
+
+
+def date_to_cftime(date, calendar):
+    """
```

Comment:
Let me know if you are aware of any built in ways to do this – I couldn't find any from a quick search.

Reply:
Judging from this `xarray` function you an just subtract one `CFTime` object from another 

https://github.com/pydata/xarray/blob/main/xarray/core/resample_cftime.py#L443-L480

And yeah the code backs that up

https://github.com/Unidata/cftime/blob/master/src/cftime/_cftime.pyx#L1063-L1064

and subtraction just returns a vanilla `timedelta`

https://github.com/Unidata/cftime/blob/master/src/cftime/_cftime.pyx#L1596

Reply:
Very glad that `cftime` does this. I'm making use of this in the `seconds_between_dates` function:
```
  delta = (date_to_cftime(end_date, calendar_str)
             - date_to_cftime(start_date, calendar_str))
```
however am manually converting from `datetime.date` to `cftime.date` to do this calculation. This is because the `start_date` and `end_date`  need to be `datetime.date` objects for other calculations such as `runtime_from_date`.